src/hotspot/share/c1/c1_Canonicalizer.cpp

```diff
@@ -845,6 +845,7 @@ void Canonicalizer::do_LookupSwitch(LookupSwitch* x) {
     for (int i = 0; i < x->length(); i++) {
       if (v == x->key_at(i)) {
         sux = x->sux_at(i);
+        break;
```

Comment:
Shouldn't we also break when `v < x->key_at(i) `, meaning no key will match?  Maybe we should consider binary search?

Reply:
Updated: break when `v < x->key_at(i)`

> Maybe we should consider binary search?
From the perspective of pursuing performance, binary search can be considered. If you agree, I can change to using binary search here.

Reply:
Yes, I'd like to see how it looks.

Reply:
Updated.